Dynatrace/dynatrace-operator
# Adds default limits for the injected init-container

# Description

If no init resource requests/limits is set and we use the CSI driver, then default resource requests/limits are set.

About the default limits:
- 30m cpu and 30Mi memory made the init container work without OOM after multiple times, however it was rather slow (took around 30 sec to get ready)
- 100m cpu and 60Mi memory made it faster by a lot (~5sec to get ready), and I increased the memory just to be safe
- so I chose:
    - requests: 30m cpu and 30Mi memory
    - limits: 100m cpu and 60Mi memory

updated the `InitResources` field to use a pointer, so we can tell if it was deliberately unset which we should respect, or was not set at all, in which case the defaults should be used  

## How can this be tested?
Deploy something that will be injected + use the csi driver, and check if the resource requests/limits are present for the init container


## Checklist
